client/app/visualizations/counter/utils.js

```diff
@@ -0,0 +1,140 @@
+import { isNumber, isFinite, toString } from 'lodash';
+import numeral from 'numeral';
+
+// TODO: allow user to specify number format string instead of delimiters only
+// It will allow to remove this function (move all that weird formatting logic to a migration
+// that will set number format for all existing counter visualization)
+function numberFormat(value, decimalPoints, decimalDelimiter, thousandsDelimiter) {
+  // Temporarily update locale data (restore defaults after formatting)
+  const locale = numeral.localeData();
+  const savedDelimiters = locale.delimiters;
+
+  // Mimic old behavior - AngularJS `number` filter defaults:
+  // - `,` as thousands delimiter
+  // - `.` as decimal delimiter
+  // - three decimal points
+  locale.delimiters = {
+    thousands: ',',
+    decimal: '.',
+  };
+  let formatString = '0,0.000';
+  if (
+    (Number.isFinite(decimalPoints) && (decimalPoints >= 0)) ||
+    decimalDelimiter ||
+    thousandsDelimiter
+  ) {
+    locale.delimiters = {
+      thousands: thousandsDelimiter,
+      decimal: decimalDelimiter || '.',
+    };
+
+    formatString = '0,0';
+    if (decimalPoints > 0) {
+      formatString += '.';
+      while (decimalPoints > 0) {
+        formatString += '0';
+        decimalPoints -= 1;
+      }
+    }
+  }
+  const result = numeral(value).format(formatString);
+
+  locale.delimiters = savedDelimiters;
+  return result;
+}
+
+// 0 - special case, use first record
+// 1..N - 1-based record number from beginning (wraps if greater than dataset size)
+// -1..-N - 1-based record number from end (wraps if greater than dataset size)
+function getRowNumber(index, rowsCount) {
```

Comment:
@arikfr I may need your opinion about this ^

Reply:
The description sounds right :)

Reply:
Previous implementation worked in a similar way (I believe), so I first described how it should behave to be clear for users, and then implemented according to that.